code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,215 @@
+	desc = "A mining charge. This one seems much more powerful than normal!"
+	icon_state = "mining-charge-3"
+	item_state = "charge_mega"
+	smoke_amount = 5
+	boom_sizes = list(4, 6, 8)
+
+/obj/item/storage/backpack/duffel/miningcharges/populate_contents()
+	for(var/i in 1 to 5)
+		new /obj/item/grenade/plastic/miningcharge/lesser(src)
+	for(var/i in 1 to 3)
+		new /obj/item/grenade/plastic/miningcharge(src)
+	new /obj/item/detonator(src)
+
+//MINING CHARGE HACKER
+
+/obj/item/t_scanner/adv_mining_scanner/syndicate
+	desc = "A scanner that automatically checks surrounding rock for useful minerals; it can also be used to stop gibtonite detonations. Wear meson scanners for optimal results. This scanner has an extra port for overriding mining charge safeties."
+
+	var/charges = 6
+
+/obj/item/t_scanner/adv_mining_scanner/syndicate/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!istype(target, /obj/item/grenade/plastic/miningcharge))
+		return
+	var/obj/item/grenade/plastic/miningcharge/charge = target
+	if(charge.hacked)
+		to_chat(user, "<span class='notice'>[src] is already overridden!</span>")
+		return
+	if(charges <= 0)
+		to_chat(user, "<span class='notice'>Its overriding function is depleted.</span>")
+		return
+	charge.override_safety()
+	visible_message("<span class='warning'>Sparks fly out of [src]!</span>")
+	playsound(src, "sparks", 50, TRUE)
+	charges--
+	if(charges <= 0)
+		to_chat(user ,"<span class='warning'>[src]'s internal battery for overriding mining charges has run dry!</span>")
+
+// MINING CHARGES DETONATOR
+
+/obj/item/detonator
+	name = "mining charge detonator"
+	desc = "A specialized mining device designed for controlled demolition operations using mining explosives."
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "Detonator-0"
+
+	/// list of all bombs connected to a detonator for a moment
+	var/list/bombs = list()
+
+/obj/item/detonator/examine(mob/user)
+	. = ..()
+	if(length(bombs))
+		. += "<span class='notice'>List of synched bombs:</span>"
+		for(var/obj/item/grenade/plastic/miningcharge/charge in bombs)
+			. += "<span class='notice'>[bicon(charge)] [charge]. Current status: [charge.installed ? "ready to detonate" : "ready to deploy"]."
+
+/obj/item/detonator/update_icon()
+	. = ..()
+	if(length(bombs))
+		icon_state = "Detonator-1"
+	else
+		icon_state = initial(icon_state)
+
+/obj/item/detonator/attack_self(mob/user)
+	playsound(src, 'sound/items/detonator.ogg', 40)
+	if(length(bombs))
+		to_chat(user, "<span class='notice'>Activating explosives...</span>")
+		for(var/obj/item/grenade/plastic/miningcharge/charge in bombs)
+			if(QDELETED(charge))
+				to_chat(user, "<span class='notice'>Can't reach [charge]. Deleting from the list...</span>")
+				bombs -= charge
+				return
+			if(charge.installed)
+				bombs -= charge
+				charge.detonate()
+	else
+		to_chat(user, "<span class='warning'>There is no charges linked to a detonator!</span>")
+	update_icon()
```

Comment:
Same here, if you simply receive the message that there are no charges then theres no point in updating the icon.